src/app/shared/courses/courses.service.ts

```diff
@@ -1,10 +1,12 @@
-import { Observable } from 'rxjs';
+import { PreloadingService } from './../preloading.service';
+import { from, Observable } from 'rxjs';
 import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { OrderByPipe } from './../../order-by.pipe';
 import { Injectable } from '@angular/core';
 import { Course } from './course.interface';
 import { mockedCourses } from './courses.mock';
 
+const ROOT_URL = 'http://localhost:3004/courses/';
```

Comment:
this might be a part of environment configuration.. so you will need to either create ConfigService or use `environment.ts` for that.